modules/storages/app/models/storages/nextcloud_storage.rb

```diff
@@ -27,22 +27,24 @@
 #++
 
 class Storages::NextcloudStorage < Storages::Storage
-  store_accessor :provider_fields,
-                 %i[username
-                    password
-                    group
-                    group_folder
-                    has_managed_project_folders]
+  PROVIDER_FIELDS_DEFAULTS = {
+    automatically_managed: true,
+    username: 'OpenProject'
+  }.freeze
 
-  alias_method :has_managed_project_folders?, :has_managed_project_folders
+  store_attribute :provider_fields, :automatically_managed, :boolean
+  store_attribute :provider_fields, :username, :string
+  store_attribute :provider_fields, :password, :string
+  store_attribute :provider_fields, :group, :string, default: PROVIDER_FIELDS_DEFAULTS[:username]
+  store_attribute :provider_fields, :group_folder, :string, default: PROVIDER_FIELDS_DEFAULTS[:username]
```

Comment:
Newly generated accessors behaves differently. A specified default value is returned only when the record is new. So, we have to decide how to handle the case, for instance, with existing records where `group` and `group_folder` have not been saved to the database. And we should make sure that when we create new records using the form `group` and `group_folder` are saved to the database or use something like previous implementation with the following logic: `provider_fields['group'] || ''OpenProject'`

Reply:
Added https://github.com/opf/openproject/pull/12721/commits/a78ca272136d2a6cadadc02a3de6ed47d491f1dd